src/Http/RequestMatcher/QuerystringRequestMatcher.php

```diff
@@ -45,6 +41,14 @@ public function matches(Request $request): bool
             return true;
         }
 
-        return (bool) preg_match('{'.$this->queryString.'}', rawurldecode($request->getQueryString() ?: ''));
+        if ($request->getQueryString()) {
+            return (bool) preg_match('{'.$this->queryString.'}', rawurldecode($request->getQueryString() ?: ''));
+        }
+
+        if ($request->getRequestUri()) {
+            return (bool) preg_match('#'.$this->queryString.'#', $request->getRequestUri());
```

Comment:
this makes the query string matcher match on request uri when there is no query string. i don't think that is correct

Reply:
also, with the check if query string is null, we should never end up here.